src/transforms/add_fields.rs

```diff
@@ -117,17 +116,21 @@ impl Transform for AddFields {
                 TemplateOrValue::Value(v) => v,
             };
             if self.overwrite {
-                if event.as_mut_log().insert(&key_string, value).is_some() {
+                if event
+                    .as_mut_log()
+                    .insert_path(key.clone().into(), value)
+                    .is_some()
+                {
                     emit!(AddFieldsFieldOverwritten {
-                        field: &format!("{}", &key),
+                        field: &format!("{}", key),
                     });
                 }
-            } else if event.as_mut_log().contains(&key_string) {
+            } else if event.as_mut_log().contains(&key.to_string()) {
```

Comment:
This forces another `.to_string()` on top of the `into()` call above and is not ideal. This is why line 105 existed. :)